src/logger.ts

```diff
@@ -2,79 +2,57 @@ import debug, { IDebugger } from "debug";
 
 type Message = string;
 
-export interface Logger {
-  enabled(): boolean;
-  trace(message: Message): void;
-  debug(message: Message): void;
-  info(message: Message): void;
-  warn(message: Message): void;
-  error(message: Message): void;
-}
+type Options = {
+  containerId?: string;
+  imageName?: string;
+};
 
-class DebugLogger implements Logger {
+export class Logger {
   private readonly logger: IDebugger;
 
-  constructor(namespace: string) {
+  constructor(namespace: string, private readonly showLevel = true) {
     this.logger = debug(namespace);
   }
 
   public enabled(): boolean {
     return this.logger.enabled;
   }
 
-  public trace(message: Message): void {
-    this.logger(`TRACE ${message}`);
-  }
-
-  public debug(message: Message): void {
-    this.logger(`DEBUG ${message}`);
-  }
-
-  public info(message: Message): void {
-    this.logger(`INFO  ${message}`);
-  }
-
-  public warn(message: Message): void {
-    this.logger(`WARN  ${message}`);
-  }
-
-  public error(message: Message): void {
-    this.logger(`ERROR ${message}`);
-  }
-}
-
-export class FakeLogger implements Logger {
-  public readonly traceLogs: Message[] = [];
-  public readonly debugLogs: Message[] = [];
-  public readonly infoLogs: Message[] = [];
-  public readonly warnLogs: Message[] = [];
-  public readonly errorLogs: Message[] = [];
-
-  public enabled(): boolean {
-    return true;
+  public trace(message: Message, options?: Options): void {
+    this.logger(`${this.showLevel ? "[TRAC] " : ""}${this.renderOptions(options)}${message}`);
   }
 
-  public trace(message: Message): void {
-    this.traceLogs.push(message);
+  public debug(message: Message, options?: Options): void {
+    this.logger(`${this.showLevel ? "[DEBU] " : ""}${this.renderOptions(options)}${message}`);
   }
 
-  public debug(message: Message): void {
-    this.debugLogs.push(message);
+  public info(message: Message, options?: Options): void {
+    this.logger(`${this.showLevel ? "[INFO] " : ""}${this.renderOptions(options)}${message}`);
   }
 
-  public info(message: Message): void {
-    this.infoLogs.push(message);
+  public warn(message: Message, options?: Options): void {
+    this.logger(`${this.showLevel ? "[WARN] " : ""}${this.renderOptions(options)}${message}`);
   }
 
-  public warn(message: Message): void {
-    this.warnLogs.push(message);
+  public error(message: Message, options?: Options): void {
+    this.logger(`${this.showLevel && "[ERRO] "}${this.renderOptions(options)}${message}`);
   }
 
-  public error(message: Message): void {
-    this.errorLogs.push(message);
+  private renderOptions(options?: Options): string {
+    let str = "";
+    if (options?.containerId) {
+      str += `[${options.containerId.substring(0, 12)}] `;
+    }
+    if (options?.imageName) {
+      str += `[${options.imageName}] `;
```

Comment:
Can you set here name of container? And if it is not exists use imageName? Name will be shorter than imageName in output.

Reply:
This field is only set when pulling an image, at which point there isn't a container